nix/lib/default.nix

```diff
@@ -279,53 +314,66 @@ in rec {
       };
     in self;
 
-  filterSource = src_tmp: { include ? [ "/" ], exclude ? [] }: src_x:
-    let
-      startsWith = s: x: builtins.substring 0 (builtins.stringLength x) s == x;
-      src = src_tmp + ("/" + src_x);
-      relativePath = path:
-        builtins.substring (builtins.stringLength (builtins.toString src))
-                           (builtins.stringLength path)
-                           path;
-    in
-      builtins.filterSource (path: type: 
-        if builtins.any (x: x) (builtins.map (startsWith (relativePath path)) exclude) then false
-        else if builtins.any (x: x) (builtins.map (startsWith (relativePath path)) include) then true
-        else false
-      ) src;
+  filterSource = src:
+    { name ? null
+    , include ? [ "/" ]
+    , exclude ? []
+    }:
+      assert name == null -> include != null;
+      assert name == null -> exclude != null;
+      let
+        _include= if include == null then [
+          "/VERSION"
+          "/${name}"
+          "/tests"
+          "/MANIFEST.in"
+          "/settings.py"
+          "/setup.py"
+        ] else include;
+        _exclude = if exclude == null then [
+            "/${name}.egg-info"
+        ] else exclude;
+        startsWith = s: x: builtins.substring 0 (builtins.stringLength x) s == x;
+        relativePath = path:
+          builtins.substring (builtins.stringLength (builtins.toString src))
+                             (builtins.stringLength path)
+                             path;
+      in
+        builtins.filterSource (path: type: 
+          if builtins.any (x: x) (builtins.map (startsWith (relativePath path)) _exclude) then false
+          else if builtins.any (x: x) (builtins.map (startsWith (relativePath path)) _include) then true
+          else false
+        ) src;
 
   mkBackend =
     { name
     , version
     , src
-    , srcs
     , python
-    , buildRequirements ? []
-    , propagatedRequirements ? []
+    , buildInputs ? []
+    , propagatedBuildInputs ? []
     , passthru ? {}
     }:
     let
-      getSrc = x:
-        if builtins.typeOf src == "path"
-        then src + ("/" + x)
-        else src x;
       self = python.mkDerivation {
         namePrefix = "";
         name = "${name}-${version}";
-        srcs = if inNixShell then null else (builtins.map getSrc srcs);
-        sourceRoot = ".";
-        buildInputs = [ makeWrapper glibcLocales ] ++
-          fromRequirementsFile buildRequirements python.packages;
-        propagatedBuildInputs =
-          fromRequirementsFile propagatedRequirements python.packages;
+
+        inherit src;
+
+        buildInputs = [
+          makeWrapper
+          glibcLocales
+          python.packages."flake8"
+          python.packages."gunicorn"
```

Comment:
i think i should add this to ``releng_common``. eg. when extra ``db`` is requested also add ``pysopg2``